server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4224,10 +4224,11 @@ public boolean scheduledRefresh() {
             if (listenerNeedsRefresh == false // if we have a listener that is waiting for a refresh we need to force it
                 && isSearchIdle()
                 && indexSettings.isExplicitRefresh() == false
-                && indexSettings.isSegRepEnabled() == false
-                // Indices with segrep enabled will never wait on a refresh and ignore shard idle. Primary shards push out new segments only
+                // Indices with segrep enabled will never wait on a refresh and ignore shard idle unless there are no replicas. Primary
```

Comment:
Annoying style nitpick, but I might rewrite this as:

```
if (listenerNeedsRefresh == false // if we have a listener that is waiting for a refresh we need to force it
    && isSearchIdleSupported()
    && isSearchIdle()
    && indexSettings.isExplicitRefresh() == false
    && active.get()) { // it must be active otherwise we might not free up segment memory once the shard became inactive
    // lets skip this refresh since we are search idle and
    // don't necessarily need to refresh. the next searcher access will register a refreshListener and that will
    // cause the next schedule to refresh.
```

The seg rep / num replicas condition moves into a new `isSearchIdleSupported()` method and (IMO) this if condition is a bit easier to read and understand.